swerve_pico/messaging-library/messaging-library.h

```diff
@@ -0,0 +1,178 @@
+#include "swerve-module-id.h"
+
+#define PAYLOAD_MAX_LEN 127
+#define HEAD_ID 0xCC
+#define TAIL_ID 0xB9
+
+enum messageParseResult {
+    MESSAGE_BUFFER_OVERFLOW = -2,
+    MESSAGE_PARSE_FAILURE,
+    MESSAGE_PARSE_NONE,
+    MESSAGE_PARSE_SUCCESS
+};
+
+enum messageReadState {
+    MESSAGE_ERROR = -1,
+    MESSAGE_HEAD,
+    MESSAGE_LENGTH,
+    MESSAGE_PAYLOAD,
+    MESSAGE_TAIL,
+    MESSAGE_CHECKSUM,
+    MESSAGE_END
+};
+
+/**
+ * BSDChecksum: Calculates checksum for messaging, using BSDChecksum algorithm. The max length is 128 bytes. The first byte is the message ID, the next 127 bytes is the payload
+ * 
+ * @param payloadString string of payload and id, 128 bytes max
+ * @param payloadLen length of payload with id
+ * 
+ * @return calculated checksum
+ */
+char MessagingBSDChecksum(char* payloadString, unsigned char payloadLen) {
```

Comment:
ok so everything above this line is fine, but actual code in the function definition (everything between the curly braces) should be in the cpp file